misc/tb/src/TB.java

```diff
@@ -0,0 +1,131 @@
+            System.exit(1);
+            return;
+        }
+
+        // Build Debezium configuration.
+
+        Properties config = new Properties();
+        config.setProperty("database.hostname", ns.getString("hostname"));
+        config.setProperty("database.port", ns.getString("port"));
+        config.setProperty("database.user", ns.getString("user"));
+        config.setProperty("database.dbname", ns.getString("database"));
+        config.setProperty("database.password", ns.getString("password"));
+        config.setProperty("database.server.name", "tb");
+        config.setProperty("database.history", "io.debezium.relational.history.FileDatabaseHistory");
+        config.setProperty("database.history.file.filename", ns.getString("save_file") + ".history");
+        config.setProperty("offset.storage", "org.apache.kafka.connect.storage.FileOffsetBackingStore");
+        config.setProperty("offset.storage.file.filename", ns.getString("save_file") + ".offsets");
+        config.setProperty("offset.flush.interval.ms", "100");
+        config.setProperty("plugin.name", "pgoutput");
+        config.setProperty("provide.transaction.metadata", "true");
+        config.setProperty("provide.transaction.metadata.file.filename", ns.getString("save_file") + ".trx");
+
+        // Need a distinct pg_replication_slots name, as "debezium" is already
+        // taken by the standard Materialize setup.
+        config.setProperty("slot.name", ns.getString("replication_slot"));
+
+        String whiteListField = ns.getString("whitelist");
+        if (whiteListField != null) {
+            config.setProperty("table.whitelist", whiteListField);
+        }
+
+        String type = ns.getString("type");
+        if (type.equals("mysql")) {
+            config.setProperty("connector.class", "io.debezium.connector.mysql.MySqlConnector");
+            config.setProperty("name", "mysql-connector");;
+        } else if (type.equals("postgres")) {
+            config.setProperty("connector.class", "io.debezium.connector.postgresql.PostgresConnector");
+            config.setProperty("name", "postgres-connector");
+        } else {
+            // The value of `type` has been validated by the argument parser.
+            throw new RuntimeException("unreachable: invalid type");
+        }
+
+        Path dir = Paths.get(ns.getString("dir"));
+        run(config, dir);
+    }
+
+    private static void run(Properties config, Path dir) throws IOException {
+        Files.createDirectories(dir);
+        logger.info("Storing data files in {}", dir);
+
+        ChangeWriter cw = new ChangeWriter(dir);
+        DebeziumEngine<RecordChangeEvent<SourceRecord>> engine = DebeziumEngine
+            .create(ChangeEventFormat.of(Connect.class))
+            .using(config)
+            .notifying(cw)
+            .build();
+
+        ScheduledExecutorService executor = Executors.newScheduledThreadPool(2);
+        executor.execute(engine);
```

Comment:
If DebeziumEngine crashes (e.g. can't connect to the server) tb keeps running. Maybe the DebeziumEngine should be wrapped so if it fails to start, System.exit(-1) can be called.

Reply:
Whoops, yup. Thanks. Fixed.